### ISSUE TITLE
Cannot examine profiles when using rayon

### ISSUE THREAD

**Issue by fzyzcjy** [rayon-rs/rayon#895]
Hi thanks for this wonderful lib! However, when using profiling tools (e.g. simpleperf for android), I cannot find out information related to my code...

Let's use this simple example:

```
use anyhow::*;
use log::*;
use rayon::prelude::*;

fn sum_of_squares(input: &[i32]) -> f64 {
    input
        .par_iter()
        .map(|&i| (i as f64).sin() * (i as f64).cos())
        .sum()
}

#[inline(never)]
#[test]
fn simple_expr_rayon() -> Result<()> {
    info!("start simple_expr_rayon");

    let vec: Vec<_> = (0..50000000).collect();
    info!("vec.len={}", vec.len());

    let value = sum_of_squares(&vec);
    info!("value={}", value);

    Ok(())
}
```

Its main thread (with offcpu profiling) is as follows. This is reasonable.

![image](https://user-images.githubusercontent.com/5236035/139001479-fdadbca2-147e-432c-a8f6-80f83722e91d.png)

However, looking at Rayon threads:

![image](https://user-images.githubusercontent.com/5236035/139001494-3236ffc8-90b3-49e3-8608-16c004dd7702.png)

another example of rayon threads:

![image](https://user-images.githubusercontent.com/5236035/139001507-70f13605-89a5-415a-92ea-b64f08a5d956.png)

I cannot see anything related to my function/my source code file. Of course, seeing `cos` I know that must be my workload. But in my realworld app, I do not have that kind of "indicators" so I am completely lost.

### In short

What I hope to have: A way to quickly realize which part of the framegraph is related to which part of *my* source code.
What I see indeed: The framegraph is filled with rayon's stack traces, and I do not know how much portion of time my workload executes.
